app/models/plan.rb

```diff
@@ -2,11 +2,25 @@ class Plan < ApplicationRecord
   before_create :generate_token
   validates :name, :price, presence: true
   belongs_to :customer
+  has_many :plan_price_history, dependent: :destroy
+  after_save :save_price_history
+
+  def list_price_history
+    plan_price_history
+      .all
+      .where.not(id: plan_price_history.last)
+      .order('created_at DESC')
+  end
 
   private
 
   def generate_token
     self.token = SecureRandom.hex(10)
     generate_token if Plan.exists?(token: token)
   end
+
+  def save_price_history
+    last_record = plan_price_history.last
+    plan_price_history.create(price: price) if last_record.nil? || last_record.price != price
```

Comment:
O que acham de usar guard clause? algo como:

```suggestion
    return if plan_price_history.last.&.price == price

    plan_price_history.create(price: price)
```

Dessa forma fica mais fácil de entender a estrutura. O operador `&` é o safe navigator ele é usado para ao chamar um método de algo nulo ou sem resposta retorne `nil` ao invés de quebrar. Então ele seria o mesmo que fazer:

`plan_price_history.last && plan_price_history.last.price` verificando se o price tem valor ou retornando nulo, e nulo não vai ser igual ao price.